Packlink/Repositories/BaseRepository.php

```diff
@@ -304,12 +304,39 @@ protected function getQueryPart(QueryCondition $condition, array $indexMap, $ali
 
         $part = "$alias.index_" . $indexMap[$column] . ' ' . $condition->getOperator();
         if (!in_array($condition->getOperator(), array(Operators::NULL, Operators::NOT_NULL), true)) {
-            $part .= " '" . IndexHelper::castFieldValue($condition->getValue(), $condition->getValueType()) . "'";
+            if (in_array($condition->getOperator(), array(Operators::NOT_IN, Operators::IN), true)) {
```

Comment:
There are no nested conditions here, only one method. It could be extracted like this:
```
$isInOperator = in_array($condition->getOperator(), array(Operators::NOT_IN, Operators::IN), true);
if ($isInOperator) {
...
```

But with this, the condition stays the same, so I would keep it as is.

Reply:
ok